source/Playnite.Common/FileSystem.cs

```diff
@@ -325,5 +326,48 @@ public static void CopyDirectory(string sourceDirName, string destDirName, bool
                 }
             }
         }
+
+        public static string LookupAlternativeFilePath(string filePath)
+        {
+            return CheckDrivesForPath(filePath, path => File.Exists(path));
+        }
+
+        public static string LookupAlternativeDirectoryPath(string directoryPath)
+        {
+            return CheckDrivesForPath(directoryPath, path => Directory.Exists(path));
+        }
+
+        private static string CheckDrivesForPath(string originalPath, Predicate<string> predicate)
+        {
+            try
+            {
+                if (!Paths.IsFullPath(originalPath))
+                {
+                    return string.Empty;
+                }
+
+                var rootPath = Path.GetPathRoot(originalPath);
+                var availableDrives = DriveInfo.GetDrives()
+                    .Where(d => d.IsReady && !d.Name.Equals(rootPath, StringComparison.OrdinalIgnoreCase));
+
+                foreach (var drive in availableDrives)
+                {
+                    var pathWithoutDrive = originalPath.Substring(drive.Name.Length);
+                    var newDirectoryPath = Path.Combine(drive.Name, pathWithoutDrive);
+                    if (predicate(newDirectoryPath))
+                    {
+                        return newDirectoryPath;
+                    }
+                }
+
+                return string.Empty;
+            }
+            catch (Exception ex) when (!Debugger.IsAttached)
+            {
+                logger.Error(ex, "Error looking for alternative path");

```

Comment:
Can you log input path please? It's a last thing, I promise :)

Reply:
ok, it's done ;-)